### PR TITLE
Both versions example.com and example.com/ are serving

### PR DESCRIPTION
**Pull request by amanjiofficial**

### CODE REVIEW

**Comment by rnehra01** [comment]
I'll review it tonight.

**Comment by amanjiofficial** [comment]
@rnehra01 okay

File: snare/cloner.py [hunk 1]
```diff
@@ -31,12 +31,22 @@ def __init__(self, root, max_depth, css_validate):
 
     @staticmethod
     def add_scheme(url):
```
Comment (glaslos) [inline snare/cloner.py]: can you add a test for this function?

File: snare/cloner.py [hunk 2]
```diff
@@ -31,12 +31,22 @@ def __init__(self, root, max_depth, css_validate):
...
+        test_url = str(yarl.URL(url))
```
Comment (rnehra01) [inline snare/cloner.py]: What is the use of this variable?

Comment (amanjiofficial) [reply snare/cloner.py]: yarl URL cannot be accessed by index so 'test_url' has string version of the URL thus we can directly go to last index and check for presence of path.

Comment (amanjiofficial) [reply snare/cloner.py]: Yes we should check conditions only for variable 'err_url' and we do not need to check it for 'new_url'.

**Comment by amanjiofficial** [comment]
I have done the required changes. Please review

File: snare/cloner.py [hunk 3]
```diff
@@ -31,11 +31,16 @@ def __init__(self, root, max_depth, css_validate):
...
     @staticmethod
     def add_scheme(url):
+        test_string = str(url)
         if yarl.URL(url).scheme:
             new_url = yarl.URL(url)
-            err_url = yarl.URL(url + '/status_404')
+            err_url = yarl.URL(url + 'status_404')
         else:
             new_url = yarl.URL('http://' + url)
+            err_url = yarl.URL('http://' + url + 'status_404')
```
Comment (rnehra01) [inline snare/cloner.py]: It will produce wrong results. Move this condition at start and just remove the trailing `/`

File: snare/cloner.py [hunk 4]
```diff
@@ -31,11 +31,16 @@ def __init__(self, root, max_depth, css_validate):
...
+        if test_string[-1] == '/':
```
Comment (rnehra01) [inline snare/cloner.py]: It will produce wrong results. Move this condition at start and just remove the trailing `/`

**Comment by amanjiofficial** [comment]
@rnehra01  Rather than removing the trailing ' _**/**_ '  I have added ' _**/**_ ' if it is not present. I have checked the code for possible cases. Please review for changes. Thank you

**Comment by amanjiofficial** [comment]
@rnehra01 pls review the code

File: snare/cloner.py [hunk 5]
```diff
@@ -31,12 +31,18 @@ def __init__(self, root, max_depth, css_validate):
 
     @staticmethod
     def add_scheme(url):
+        test_string = str(url)
```
Comment (rnehra01) [inline snare/cloner.py]: `url` is already a string, right?

File: snare/cloner.py [hunk 6]
```diff
@@ -31,12 +31,18 @@ def __init__(self, root, max_depth, css_validate):
...
+        if test_string[-1] == '/':
+            b_url = yarl.URL(url)
+        else:
+            b_url = yarl.URL(url + '/')
+        url = b_url.human_repr()
```
Comment (rnehra01) [inline snare/cloner.py]: Keep the url as a string, no need of conversion to URL object and then just add the trailing space.

**Pull request merged**